recipes-bsp/imx-mkimage/imx-boot_0.2.bb

```diff
@@ -86,7 +86,7 @@ compile_mx8m() {
     cp ${DEPLOY_DIR_IMAGE}/${BOOT_TOOLS}/${UBOOT_DTB_NAME}   ${BOOT_STAGING}
     cp ${DEPLOY_DIR_IMAGE}/${BOOT_TOOLS}/u-boot-nodtb.bin-${MACHINE}-${UBOOT_CONFIG} \
                                                              ${BOOT_STAGING}/u-boot-nodtb.bin
-    cp ${STAGING_DIR_NATIVE}/${bindir}/mkimage               ${BOOT_STAGING}/mkimage_uboot
+    ln -sf ${STAGING_DIR_NATIVE}/${bindir}/mkimage            ${BOOT_STAGING}/mkimage_uboot
```

Comment:
@otavio sorry I don't quite understand the review feedback. The mkimage tool is already built by u-boot-mkimage-native which this recipe depends on. So the binary of the tool is available in the imx-boot/0.2-r0/recipe-sysroot-native/usr/bin/mkimage. The problem is the imx-boot Makefile executes it from current working directory and with a difference name (see here https://github.com/Freescale/imx-mkimage/blob/master/iMX8M/soc.mak#L57)

Which I guess is why the original author was copying in the binary. As far as I can see only other way to fix this would be to carry a patch in meta-freescale for imx-boot to update the Makefile to remove './' and also update the name to 'mkimage' from 'mkimage_uboot'? Apologies in advance if I'm misunderstanding something here.

Reply:
You are right; but considering that, I am wondering why the copy did not work.

Did you run it inside a devshell to check why it fails?

Reply:
Check https://github.com/Freescale/meta-freescale/pull/382/commits/0cea1a55f97e17f93973ec1c91e972d9c8586c83

In fact it explains it.

Reply:
@otavio ah yes setting LD_LIBRARY_PATH to search STAGING_LIBDIR_NATIVE is another way to solve this. I guess it's up to you which "fix" you prefer / hack you dislike the least ;-)